utils/create_dummy_models.py

```diff
@@ -544,32 +571,61 @@ def convert_processors(processors, tiny_config, output_folder, result):
             )
             fast_tokenizer = None
 
-    # Make sure the slow tokenizer (if any) corresponds to the fast version (as it might be converted above)
-    if fast_tokenizer:
-        try:
-            slow_tokenizer = AutoTokenizer.from_pretrained(output_folder, use_fast=False)
-        except Exception:
-            result["warnings"].append(
-                (
-                    f"Failed to load the slow tokenizer saved from {fast_tokenizer.__class__.__name__}.",
-                    traceback.format_exc(),
-                )
+    # sanity check: fast and slow tokenizers should be compatible
+    if fast_tokenizer is not None and slow_tokenizer is not None:
+        if fast_tokenizer.vocab_size != slow_tokenizer.vocab_size:
+            warning_messagae = (
+                f"The fast/slow tokenizers have different vocabulary size: "
+                f"{fast_tokenizer.__class__.__name__}.vocab_size = {fast_tokenizer.vocab_size} and "
+                f"{slow_tokenizer.__class__.__name__}.vocab_size = {slow_tokenizer.vocab_size}."
+                "The original tokenizers from the Hub will be used instead."
             )
-            # Let's just keep the fast version
+            result["warnings"].append(warning_messagae)
+            # Let's use the original version at the end (`original_fast_tokenizer` and `original_slow_tokenizer`)
+            fast_tokenizer = None
             slow_tokenizer = None
 
-    # If the fast version can't be created and saved, let's use the slow version
-    if not fast_tokenizer and slow_tokenizer:
-        try:
-            slow_tokenizer.save_pretrained(output_folder)
-        except Exception:
-            result["warnings"].append(
-                (
-                    f"Failed to save the slow tokenizer for {slow_tokenizer.__class__.__name__}.",
-                    traceback.format_exc(),
+    # If there is any conversion failed, we keep the original tokenizers
+    if (original_fast_tokenizer is not None and fast_tokenizer is None) or (
+        original_slow_tokenizer is not None and slow_tokenizer is None
+    ):
+        warning_messagae = (
+            "There are some issues when converting the fast/slow tokenizers. The original tokenizers from the Hub "
+            " will be used instead."
+        )
+        result["warnings"].append(warning_messagae)
+        # Let's use the original version at the end (`original_fast_tokenizer` and `original_slow_tokenizer`)
+        fast_tokenizer = original_fast_tokenizer
+        slow_tokenizer = original_slow_tokenizer
```

Comment:
Make sure we are able to get tiny versions for all tokenizers involved. If not, use the original ones.